hw/ip/gpio/data/gpio.hjson

```diff
@@ -390,7 +391,14 @@
       ],
     },
     { name: "HW_STRAPS_DATA_IN",
-      desc: "GPIO Input data sampled as straps during cold boot read value",
+      desc: '''
+        GPIO input data that was sampled as straps as the block came out of reset.
+
+        This register depends on the GpioAsHwStrapsEn parameter.
+        - If the parameter is false then the register reads as zero.
+        - If the parameter is true then GPIO input data is sampled on the first cycle after reset
+          where the strap_en_i input is high. That sampled data is stored in this register.
```

Comment:
I'd suggest reordering this to "... is sampled after reset on the first cycle where the strap_en_i input is high."

That avoids it sounding like we only do this for special sorts of reset.